src/components/screens/consul/Proposals/NewProposal.js

```diff
@@ -129,10 +146,42 @@ export const NewProposal = ({ navigation, data, query }) => {
         tagList: newProposalData?.tagList,
         termsOfService: hasAcceptedTermsOfService,
         videoUrl: newProposalData?.videoUrl
+
+        // TODO: image and document upload needed here? and if yes, how to do it?
+        //  documentsAttributes: [
+        // 	  {
+        // 		  title: 'sample.pdf',
+        // 		  cachedAttachment:
+        // 		  	'/Users/ardasenturk/Development/SVA/consul-bb/public/system/documents/cached_attachments/user/49/original/f2245afb48be5f906474ad929aacb7f91f408a23.pdf'
+        // 	  }
+        //  ],
       }
     };
 
-    if (!hasAcceptedTermsOfService) return showRegistrationFailAlert();
+    if (!hasAcceptedTermsOfService) return showPrivacyCheckRequireAlert();
+
+    if (newProposalData.image) {
+      const imageUploadData = await uploadAttachment(newProposalData.image, 'image', 'Proposal');
+
+      if (imageUploadData.status === 200) {
+        imageData = JSON.parse(imageUploadData.body);
+
+        let imageAttributes = {
+          title: imageData?.filename,
+          cachedAttachment: imageData?.cached_attachment
+        };
+
+        variables.attributes = { ...variables.attributes, imageAttributes };
+      } else if (imageUploadData.status === 422) {
+        if (JSON.parse(imageUploadData.body).errors.length < 100) {
```

Comment:
what does the `100` stand for? how does the errors look like in that case?

Reply:
i see. this seems not very long lasting for me.
if there are errors, we should use them for translation.
one possibility would be to lowercase and hyphenate the string:

```
const errors = "Choose image content type image/png does not match any of accepted content types jpg";

errors.toLowerCase().split(" ").join("-")
=> "choose-image-content-type-image/png-does-not-match-any-of-accepted-content-types-jpg"
```

with such string we could list it in `texts` and use it as follows:

```
'choose-image-content-type-image/png-does-not-match-any-of-accepted-content-types-jpg': 'Bildtyp image/png entspricht nicht dem akzeptierten Typen jpg'
```

Reply:
Yes, this is a logical method. 👍